src/index.ts

```diff
@@ -1 +1,5 @@
-export * from "./mochaPlugin/index.js";
+import {suite, test} from "@vitest/runner";
+
+export {bench, itBench, setBenchOpts, setBenchmarkOptions} from "./benchmark/index.js";
+export const describe = suite;
+export const it = test;
```

Comment:
Do we need to export `it`.  Should we only expose the `bench` and `itBench`??

Reply:
In few examples and usages, we have test suits which have mix of bench and normal tests. So we need to export `it`.

Reply:
Silly question, but if we are using `@vitest/runner` is that the same functionality for `it` as would be expected from a standard test suite?

Reply:
Yes it is but then you have to add `vitest` in your dependencies along with  the `@chainsafe/benchmark` package.

Reply:
I think the consumer should import `it` from the testing library then.  Seems a better pattern.

Reply:
What if they don't have any? What if they have some other test runner? Ideally in benchmark tests they should not mix the other tests. For now it's for the compatibility to run such tests which had a mix. We can mark it deprecated and remove it in later releases.

Reply:
I disagree with this.  If they have another test runner, that is the one they should use.  If your concern is compatibility make it a breaking change and remove it.  If this lib is only for `bench` and `benchIt` it does not need to support anything else, that is up to the consumer.

Reply:
And second thought if they have an `it` in an existing benchmark then this will be breaking anyway because its not using Mocha and the old one was.  It should go and the consumer should explicitly import he `it` they want to use

Reply:
> I disagree with this. If they have another test runner, that is the one they should use. If your concern is compatibility make it a breaking change and remove it. If this lib is only for `bench` and `benchIt` it does not need to support anything else, that is up to the consumer.

This is not a lib, it's a benchmarking tool comes with it's own binary to run. And for such files one should not mix up the DSL from other test runners.

Reply:
> And second thought if they have an `it` in an existing benchmark then this will be breaking anyway because its not using Mocha and the old one was. It should go and the consumer should explicitly import he `it` they want to use

Yes that will be a breaking change on import syntax, not the API level of the `it` itself.

Reply:
As we don't run Jasmine test code with Vitest or neither Vitest test code with Jasmine, or similarly to other runners. Same way one should not run the benchmarking test files with tools other than `benchmark` itself.

Reply:
I agree with all of that.  It also argues, correctly, for removing the export....  It should not part of the `benchmark` API that is exported

Reply:
`describe` is different because it is for grouping suites of benchmarks.... `test` is for only one thing... test cases with `expect`

Reply:
> I agree with all of that. It also argues, correctly, for removing the export.... It should not part of the `benchmark` API that is exported

Yes it should not **ideally**, exported here just for people to only change the import and nothing else after this release. and I plan to mark it deprecated and remove in coming releases.

Reply:
@wemeetagain @nflaig what are your thoughts on this?

Reply:
I don't understand why this package should export `it`, I don't see use using it inside Lodestar either, what's the use case?

And are we worried about backward compatiblity? as I understand we want to publish this as `@chainsafe/benchmark` which would be a new package

Reply:
This library is really tied to vitest/runner. I think how/if we go about re-exporting from vitest depends on what the library is trying to achieve.
Is this an extension to vitest to support benchmarks (where consumers can use vitest features + our benchmark features)?
Or is this an opinionated benchmarking tool with an API that merely mirrors a subset of vitest?

If its a vitest extension, maybe we shouldn't re-export anything from vitest runner (vitest/runner treated as a peer dependency) (or alternatively, we should re-export everything (vitest/runner treated as a dependency)).
If its an opinionated benchmarking tool, we should just re-export what we specifically need.

Also, we want `describe, do we want to support `afterEach`/`afterAll`, etc?

Reply:
> Is this an extension to vitest to support benchmarks (where consumers can use vitest features + our benchmark features)?

Vitest have it's own (benchmarking feature)[https://vitest.dev/guide/features.html#benchmarking] and our tool does not have anything to do do with it. 

> Or is this an opinionated benchmarking tool with an API that merely mirrors a subset of vitest?

Yes it's an independant tool which have a DSL for `describe` and `it` similar to vitest as well as mocha. 

> If its a vitest extension, maybe we shouldn't re-export anything from vitest runner

If its a vitest extension, maybe we shouldn't re-export anything from vitest runner
No it's not a vitest extension, if it was then it would not have it's own binary to run. 

Also, we want describe, do we want to support afterEach/afterAll, etc?
We have similar callbacks supported in `itBench` options as `before` and `beforeEach`.

Reply:
For benchmarks you specify `before` or `beforeEach` as the options in `itBench`. It's the syntax from start for this library. 

https://github.com/ChainSafe/ssz/blob/b85c774a2ed2a093ab36209aa3988feee56d8fcc/packages/persistent-merkle-tree/test/perf/hasher.test.ts#L38

https://github.com/ChainSafe/ssz/blob/b85c774a2ed2a093ab36209aa3988feee56d8fcc/packages/persistent-merkle-tree/test/perf/hasher.test.ts#L42

And probable reason is that for bench marks you have specific actions for each benchmark test. And afterEach is not required bas everything is discarded anyway after the benchmark run.